MediaBrowser.Controller/Lyrics/LyricInfo.cs

```diff
@@ -0,0 +1,29 @@
+using System.IO;
+
+namespace MediaBrowser.Controller.Lyrics;
+
+/// <summary>
+/// Lyric helper methods.
+/// </summary>
+public static class LyricInfo
+{
+    /// <summary>
+    /// Gets matching lyric file for a requested item.
+    /// </summary>
+    /// <param name="lyricProvider">The lyricProvider interface to use.</param>
+    /// <param name="itemPath">Path of requested item.</param>
+    /// <returns>Lyric file path if passed lyric provider's supported media type is found; otherwise, null.</returns>
+    public static string? GetLyricFilePath(ILyricProvider lyricProvider, string itemPath)
+    {
+        foreach (string lyricFileExtension in lyricProvider.SupportedMediaTypes)
+        {
+            var lyricFilePath = Path.ChangeExtension(itemPath, lyricFileExtension);
+            if (File.Exists(lyricFilePath))
```

Comment:
That's potentially 5 IO calls for every music file. I wonder if it would be better to use [`Directory.GetFiles`](https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.getfiles?view=net-6.0#system-io-directory-getfiles(system-string-system-string)) eg.

```C#
            foreach (var filename in Directory.GetFiles(Path.GetDirectoryName(lyricFilePath), $"{Path.GetFilenameWithoutExtension(lyricFilePath)}.*")
```

Reply:
I agree. Done.